tests/test_json.py

```diff
@@ -49,6 +50,23 @@ def test_encoding(input, output):
     assert output == json.dumps(input, default=pydantic_encoder)
 
 
+def test_path_encoding(tmpdir):
+    class PathModel(BaseModel):
+        path: Path
+        file_path: FilePath
+        dir_path: DirectoryPath
+
+    tmpdir = Path(tmpdir)
+    path = tmpdir / 'foo'
+    file_path = tmpdir / 'bar'
+    file_path.touch()
+    dir_path = tmpdir / 'baz'
+    dir_path.mkdir()
+    model = PathModel(path=path, file_path=file_path, dir_path=dir_path)
+    expected = '{{"path": "{}", "file_path": "{}", "dir_path": "{}"}}'.format(path, file_path, dir_path)
```

Comment:
sorry, this isn't quite what I meant. You're still effectively checking `str(x) == str(x)` which only checks than an exception isn't raised.

Something like

```py
class PathModel(BaseModel):
    path: Path
model = PathModel(path=Path('/path/test/example/'))
assert json.dumps(model, default=pydantic_encoder) == '{"path": "/path/test/example/"}'
```

Reply:
1. I can verify that this test does verify the changes that I made. Without those changes to `ENCODERS_BY_TYPE`, the test fails.
2. When I actually use a `Model` like `PathModel`, I won't be passing in a `Path` object, i'll be passing in a string, from a json-deserialized dict.
3. We can instantiate `pathlib.Path` objects, but we can't instantiate `FilePath` and `DirectoryPath` objects.  (It's for this reason that I wrote a new test, and didn't add a parameter to `test_encoding`.) See https://stackoverflow.com/questions/29850801/subclass-pathlib-path-fails and below:

```py
>>> import pydantic
>>> pydantic.DirectoryPath('/')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<snip>/pydantic/lib/python3.7/pathlib.py", line 1003, in __new__
    self = cls._from_parts(args, init=False)
  File "<snip>/pydantic/lib/python3.7/pathlib.py", line 658, in _from_parts
    drv, root, parts = self._parse_args(args)
  File "<snip>/pydantic/lib/python3.7/pathlib.py", line 651, in _parse_args
    return cls._flavour.parse_parts(parts)
AttributeError: type object 'DirectoryPath' has no attribute '_flavour'
```

4. `FilePath` and `DirectoryPath` require the file and directory to exist, so that's why I'm using `tmpdir` and actually creating a file and a directory.   Using the `tmpdir` fixture means the location of `file_path` and `dir_path` are not static.
5. If you'd like to only test `Path`, I can do that with a static value and the test will look like how you suggest.